parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -38,6 +38,12 @@
 
 abstract public class Binary implements Comparable<Binary>, Serializable {
 
+  public interface ComparatorHelper {
```

Comment:
These can be static methods that create correct `ByteBuffers` and then call the right `Comparator<ByteBuffer>`. No need to define an interface.

Reply:
Currently, we have 4 different `Binary` implementations with 2 inner representations: `byte[]` and `ByteBuffer`. So, we need 3 different comparisons: `byte[]` to `byte[]`, `ByteBuffer` to `ByteBuffer` and `byte[]` to `ByteBuffer`.
The concept here with this interface is to make it able to have `Comparator<Binary>` implemented outside of this class. 
I would even implement it similarly if we want to have different `Binary` instances with the proper `Comparable<Binary>` implementation only in this case the interface would be private.
Meanwhile, I would vote on having separate `Comparator<Binary>` implementations in the same place where the other primitive comparators are.
What do you think?

Reply:
My point is that this doesn't need to be a class because these are straight-forward transformations to `compare(ByteBuffer, ByteBuffer)`. The `byte[]` arguments can be easily wrapped, and the `ByteBuffer` args should duplicate the buffer and then set the duplicate's `position` and `limit` appropriately.

Reply:
If I understand it correctly, it means that a `byte[]` backed `Binary` shall either create new `ByteBuffer` instances to wrap the `byte[]` references for each `compare(Binary, Binary)` invocation or have a `ByteBuffer` reference cached inside the `Binary` implementation.
The first one would impact the performance of filtering and creating the statistics the latter one would increase the memory footprint of the binary representation.
What do you think?

Reply:
Yes, what I'm suggesting would mean wrapping the byte array in a buffer to send it into compare, but I don't think avoiding object creation here is going to be worth it for performance. I think the main concern is to make sure we have as few code paths for this comparison as possible for correctness.

Reply:
Currently, we have 5 different implementations for the abstract class `Binary`. I guess, it is for performance reasons.
If we really think wrapping a `byte[]` into a `ByteBuffer` for comparing `Binary` object worth for the clean code then I would suggest completely refactoring the class `Binary` to have only one implementation which contains a `ByteBuffer`.

Reply:
There's quite a bit of refactoring in #390 for Binary to use ByteBuffer more. Lets see where we're at when those two are in.